### PR TITLE
Reorder Helix elements for Isolate Geometry Mode to display isolated geometry

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

https://jira.autodesk.com/browse/DYN-3130

This attempts to fix a regression with Isolate Selected Geometry feature where point-line geometry when selected does not appear when hidden by mesh geometry (solids and surfaces). The reason for this is that the geometry needs to be reordered so that selected geometries will have higher precedence and rendered closer to the camera compared to unselected geometries.

Caveat: This comes with some performance penalty when rendering selected geometry in isolated mode. 
isolate_pointline_geom

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 1]
```diff
@@ -523,6 +523,28 @@ protected HelixWatch3DViewModel(NodeModel model, Watch3DViewModelStartupParams p
...
+            if (Camera != null)
+            {
+                values.Sort(new Element3DComparer(Camera.Position));
+            }
+
+            if (sceneItems == null)
+                sceneItems = new ObservableElement3DCollection();
+
+            sceneItems.Clear();
+            sceneItems.AddRange(values);
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs:547]: This logic has been lifted off of `UpdateSceneItems`, however, `UpdateSceneItems` seems to be doing some extra work like cycling through the list of elements and updating vertices, triangles, and colors, which I'm not sure if that is necessary in this case. Skipping that saves some time.

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 2]
```diff
@@ -523,6 +523,28 @@ protected HelixWatch3DViewModel(NodeModel model, Watch3DViewModelStartupParams p
 
             SetupScene();
             InitializeHelix();
+            DynamoSelection.Instance.Selection.CollectionChanged += OnSelectionCollectionChanged;
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs:526]: I think we should find an appropriate spot to unsubscribe this handler.

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 3]
```diff
@@ -523,6 +523,28 @@ protected HelixWatch3DViewModel(NodeModel model, Watch3DViewModelStartupParams p
...
+            if (Element3DDictionary == null)
+            {
+                sceneItems = new ObservableElement3DCollection();
+                return;
+            }
+
+            var values = Element3DDictionary.Values.ToList();
+            if (Camera != null)
+            {
+                values.Sort(new Element3DComparer(Camera.Position));
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs:540]: Why does this need to happen every time the selection changes? Couldn't it be done once when the isolation mode is enabled?

Comment (aparajit-pratap) [reply src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: Tried that. Doesn't work for some reason.

**Comment by mjkkirschner** [comment]
nice! @aparajit-pratap 

would you also mind adding an image comparison test for this so it does not regress again? I added a few for the mesh mesh case in the last pr.

https://github.com/DynamoDS/Dynamo/commit/89818c3e9b1dd8a019980546e4c13793b9229c78

**Review by mjkkirschner — APPROVED** [review]

**Pull request merged**